rust/processor/src/db/common/models/emojicoin_models/models/market_latest_state_event.rs

```diff
@@ -0,0 +1,151 @@
+                maybe_tracker_1d = Some(tracker);
+            }
+        });
+
+        let tracker_1m = maybe_tracker_1m.expect("Every market should have a PERIOD_1M tracker.");
+        let tracker_1d = maybe_tracker_1d.expect("Every market should have a PERIOD_1D tracker.");
+
+        Self {
+            transaction_version: txn_info.version,
+            sender: txn_info.sender,
+            entry_function: txn_info.entry_function,
+            transaction_timestamp: txn_info.timestamp,
+
+            market_id: metadata.market_id,
+            symbol_bytes: metadata.emoji_bytes,
+            bump_time: micros_to_naive_datetime(sequence_info.last_bump_time),
+            market_nonce: sequence_info.nonce,
+            trigger,
+
+            clamm_virtual_reserves_base: clamm_virtual_reserves.base,
+            clamm_virtual_reserves_quote: clamm_virtual_reserves.quote,
+            cpamm_real_reserves_base: cpamm_real_reserves.base,
+            cpamm_real_reserves_quote: cpamm_real_reserves.quote,
+            lp_coin_supply,
+            cumulative_stats_base_volume: cumulative_stats.base_volume,
+            cumulative_stats_quote_volume: cumulative_stats.quote_volume,
+            cumulative_stats_integrator_fees: cumulative_stats.integrator_fees,
+            cumulative_stats_pool_fees_base: cumulative_stats.pool_fees_base,
+            cumulative_stats_pool_fees_quote: cumulative_stats.pool_fees_quote,
+            cumulative_stats_n_swaps: cumulative_stats.n_swaps,
+            cumulative_stats_n_chat_messages: cumulative_stats.n_chat_messages,
+            instantaneous_stats_total_quote_locked: instant_stats.total_quote_locked,
+            instantaneous_stats_total_value_locked: instant_stats.total_value_locked,
+            instantaneous_stats_market_cap: instant_stats.market_cap,
+            instantaneous_stats_fully_diluted_value: instant_stats.fully_diluted_value,
+            last_swap_is_sell: last_swap.is_sell,
+            last_swap_avg_execution_price_q64: last_swap.avg_execution_price_q64,
+            last_swap_base_volume: last_swap.base_volume,
+            last_swap_quote_volume: last_swap.quote_volume,
+            last_swap_nonce: last_swap.nonce,
+            last_swap_time: micros_to_naive_datetime(last_swap.time),
+
+            daily_tvl_per_lp_coin_growth_q64: calculate_tvl_growth(tracker_1d),
+            in_bonding_curve: tracker_1m.starts_in_bonding_curve
+                || tracker_1m.ends_in_bonding_curve,
+            volume_in_1m_state_tracker: tracker_1m.volume_quote,
+        }
+    }
+}
+
+pub fn calculate_tvl_growth(tracker_1d: PeriodicStateTracker) -> BigDecimal {
+    let PeriodicStateTracker {
+        tvl_to_lp_coin_ratio_start: start,
+        tvl_to_lp_coin_ratio_end: end,
+        ..
+    } = tracker_1d;
+
+    let a = start.tvl;
+    let b = start.lp_coins;
+    let c = end.tvl;
+    let d = end.lp_coins;
+
+    // Copied directly from the original implementation.
```

Comment:
Not exactly: Q64 multiplication requires division at the end, I don't see that here

Reply:
Whoops, I think the field name is a misnomer, too, but my comment was intended to say that the logic I used was copied directly from the Q64 implementation just without the bit shift, since we can calculate and store the decimal value directly here, and I believe it's actually more accurate with `BigDecimal`.

I checked against the outputs of what the state event emits (after calculation) vs. what `NUMERIC` stores without the q64 bit shift and the latter seemed more accurate - that would make sense, right? I figured we'd prefer to have more accuracy here since the Q64 representation was mostly just a way to represent the float in Move.

But let me know what you think, my understanding could be wrong. If it's correct though I can update the field to not have `_q64` and we could get more accurate data for small tvl growth delta

Reply:
> since we can calculate and store the decimal value directly here

As I understand that would require shifting by 64 twice: for reference, see https://github.com/econia-labs/econia-v5/blob/main/src/move/research/uniswap-v3-book-milestone-1/README.md (you might need to view the LaTeX locally because I'm noticing GitHub's render style has changed slightly)

Basically depending on what operation you do in fixed point, you need to account for how that operation affects the scalar. For example with square roots, the shift factor gets halved: https://www.cse.scu.edu/~dlewis/book3/labs/Lab11E.pdf